### PR TITLE
implement project phase 2 requirements

### PR DESCRIPTION
**Pull request by admirerbrown**
i performed the following activities on the project:

**Project requirements**
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property completed set to false 
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,33 +1,133 @@
...
-    discription: 'implement project code review suggestions',
-    completed: false,
-    index: 2,
-  },
-];
+class TaskManager {
+  constructor(taskName) {
+    this.taskName = taskName;
+    this.completed = false;
+    this.itemIndex = 0;
```
Comment (DammyShittu) [inline src/index.js]: - [x] All your items have an index value of 0. But that should not be so. Kindly note that each item index should correspond with its position in the array. So the first item should have an index of 1, the second item should have an index of 2, the third item index of 3, and so on. Kindly fix this.

  image

File: src/index.js [hunk 2]
```diff
@@ -1,33 +1,133 @@
...
+    // get all list todo's not deleted
+    oldTodoList.filter((item) => {
+      if (item.taskName !== task) {
+        currentTodoList.push(item);
+      }
+      return item;
+    });
+    localStorage.setItem('todos', JSON.stringify(currentTodoList));
+    location.reload();
+  };
```
Comment (DammyShittu) [inline src/index.js]: - [x] When a task is removed the index values of the remaining items should be updated accordingly. So if I have four items with index values of 1, 2, 3, and 4 respectively, and I remove the third item, the new index values should be 1, 2, 3 - not 1, 2, 4. Kindly implement that as well.

File: src/index.js [hunk 3]
```diff
@@ -1,33 +1,133 @@
...
+
+  // get oldTodoList from localStorage and update appropriate todo task.
+  updateTask = (newText, todoIndex) => {
+    const oldTodoList = JSON.parse(localStorage.getItem('todos'));
+    oldTodoList[todoIndex].taskName = newText;
+    localStorage.setItem('todos', JSON.stringify(oldTodoList));
+    location.reload();
+  };
+}
+
```
Comment (DammyShittu) [inline src/index.js]: - [x] According to the requirements, the functionalities for adding, deleting, and editing tasks should be in a new javascript. Kindly move all these functions to a new file and keeps this `index.js` file neater.

File: src/index.js [hunk 4]
```diff
@@ -1,33 +1,133 @@
...
+    dragBtn[i].style.display = 'none';
+    delBtn[i].style.display = 'block';
+    if (itemIndex.length === 2) {
+      contentClick[itemIndex[0]].style.backgroundColor = '';
+      dragBtn[itemIndex[0]].style.display = 'block';
+      delBtn[itemIndex[0]].style.display = 'none';
+      itemIndex = itemIndex.slice(1);
+    }
+    if (newNote.addEventListener('click', () => {
+      // console.log('yeah won');
```
Comment (DammyShittu) [inline src/index.js]: - [x] Kindly remove this commented code.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @admirerbrown,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/add_todo.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      const newTask = new TaskManager(task.value, noteIndex);
+      getTaskList.push(newTask);
+
+      // send todoList to localStorage
+      newTask.addTask(getTaskList);
+      task.value = '';
+    }
+  });
+};
+
```
Comment (youmari) [inline src/modules/add_todo.js]: - [x] as you can see in the picture below the index start on 0 but it should be start from 1 as required so please try do that in order to complete the requirements 
- Screenshot of the requirements
Screenshot 2022-12-08 at 13 26 12
Screenshot 2022-12-08 at 13 25 13

**Review by youmari — CHANGES_REQUESTED** [review]
@admirerbrown , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**